### ISSUE TITLE
mesa-vulkan-icd-wrapper D9300 Mali G720 成功运行

### ISSUE THREAD

**Issue by SCP68** [xMeM/vulkan-wsi-layer#40]
https://github.com/user-attachments/assets/2f70b251-c6f7-4c7b-a5ab-e71b68e47cae

但是仍然会出现画面跳帧 或者画面割裂

However, frame skipping or fragmentation still occurs

**Comment by SCP68** [comment]
It takes a little time to slow down or reduce the fragmentation and frame skipping.

**Comment by SCP68** [comment]
You need to turn this on for a little longer to slowly reduce the frame skipping and fragmentation

**Comment by hansm629** [comment]
@SCP68 
Is this a D3D9 demo via DXVK?

I wonder if the D3D11 demo with BC textures also works well.

**Comment by SCP68** [comment]
> @SCP68 这是通过 DXVK 进行的 D3D9 演示吗？
> 
> 我想知道带有BC纹理的D3D11演示是否也能运行良好。

I was talking to them in some groups and they said that MaliG720 would still have rendering errors in version 1.10.3. (I used mesa 24.3 for Trass 3r and still had rendering problems.

**Comment by SCP68** [comment]
I talked to them in some groups and they said that MaliG720 would still have rendering errors in version 1.10.3. (I used mesa 24.3 for Trass 3r and still had rendering problems

Specific members of the communication group said that the Breguet version of the merger dxvk 2.5.1 estimates to solve

**Comment by SCP68** [comment]
> @SCP68 这是通过 DXVK 进行的 D3D9 演示吗？
> 
> 我想知道带有 BC 纹理的 D3D11 演示是否也能运行良好。

I talked to them in some groups and they said that MaliG720 would still have rendering errors in version 1.10.3. (I used mesa 24.3 for Trass 3r and still had rendering problems

I think D 9300 Mali G 720 combineD with dxvk 2.5.1 estimates to resolve

Then as for the BC texture you mentioneD maybe MaliGPU can only Do D 3D 9 via DXVK anD no way to Do D 3D 11

**Comment by Trass3r** [comment]
> I wonder if the D3D11 demo with BC textures also works well.

Mali doesn't support BC, not even in the G720: https://vulkan.gpuinfo.org/displayreport.php?id=33753#features

DXVK 2.x does not work as-is because of missing VK_EXT_robustness2, in particular nullDescriptor.

PanVK doesn't have it yet either, only [basic robustness](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32331).

**Comment by SCP68** [comment]
> > 我想知道带有 BC 纹理的 D3D11 演示是否也能运行良好。
> 
> Mali 不支持 BC，甚至 G720 也不支持：[https://vulkan.gpuinfo.org/displayreport.php? id=33753#features](https://vulkan.gpuinfo.org/displayreport.php?id=33753#features)
> 
> 由于缺少 VK_EXT_robustness2（尤其是 nullDescriptor），DXVK 2.x 无法按原样工作。
> 
> PanVK 还没有这个功能，只有[基本的稳健性](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32331)。

So do you think the follow-up will solve these problems? At least as it is 😌

**Comment by SCP68** [comment]
> > 我想知道带有 BC 纹理的 D3D11 演示是否也能运行良好。
> 
> Mali 不支持 BC，甚至 G720 也不支持：[https://vulkan.gpuinfo.org/displayreport.php? id=33753#features](https://vulkan.gpuinfo.org/displayreport.php?id=33753#features)
> 
> 由于缺少 VK_EXT_robustness2（尤其是 nullDescriptor），DXVK 2.x 无法按原样工作。
> 
> PanVK 还没有这个功能，只有[基本的稳健性](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/32331)。

So do you think the follow-up will solve these problems? At least as it is 😌

**Comment by hansm629** [comment]
@SCP68 
Unfortunately, it doesn't support the latest Dimensity 9400 GPU either. ARM doesn't seem to be very interested in this area.

**Comment by SCP68** [comment]
> @SCP68 Unfortunately, it doesn't support the latest Dimensity 9400 GPU either. ARM doesn't seem to be very interested in this area.

Okay but will there be or support after Dimensity 9300

**Comment by SCP68** [comment]
> @SCP68 Unfortunately, it doesn't support the latest Dimensity 9400 GPU either. ARM doesn't seem to be very interested in this area.

OK but will there be or support after the D9300

**Comment by hansm629** [comment]
@SCP68 
No. If I understand correctly, ARM's future Mali GPUs will not support this `feature & extensions`.

**Comment by SCP68** [comment]
> @SCP68 不，如果我理解正确的话，ARM 未来的 Mali GPU 将不支持此功能`feature & extensions`。



> @SCP68 No. If I understand correctly, ARM's future Mali GPUs will not support this `feature & extensions`.

Okay thank you for the answers

**Comment by Trass3r** [comment]
They only added the other missing pieces like dualSrcBlend, logicOp, depthClipEnable, vertexAttributeInstanceRateDivisor

https://vulkan.gpuinfo.org/compare.php?devices[]=Xiaomi%202407FRK8EC&os=android&devices[]=vivo%20V2405A&os=android#features

**Comment by czzzc-chi** [comment]
> They only added the other missing pieces like dualSrcBlend, logicOp, depthClipEnable, vertexAttributeInstanceRateDivisor
> 
> [https://vulkan.gpuinfo.org/compare.php?devices[]=Xiaomi%202407FRK8EC&os=android&devices[]=vivo%20V2405A&os=android#features](https://vulkan.gpuinfo.org/compare.php?devices%5B%5D=Xiaomi%202407FRK8EC&os=android&devices%5B%5D=vivo%20V2405A&os=android#features)

![Screenshot_20241209_035843](https://github.com/user-attachments/assets/64f6db14-98ae-47cb-9e91-332dc68bd14a)
termux输入vulkaninfo输出的值，天玑9300，iQOONeo9Pro，vulkan1.3，很显然天玑应该有这个扩展特性，你们可以尝试为天玑适配一下dxvk2.x

**Comment by Trass3r** [comment]
Which driver version?

**Comment by czzzc-chi** [comment]
Android15,&nbsp; &nbsp;CN OriginOS5



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年12月9日(周一) 凌晨4:28
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [xMeM/vulkan-wsi-layer] mesa-vulkan-icd-wrapper D9300 Mali G720 成功运行 (Issue #40)




 
Which driver version?

**Comment by SCP68** [comment]
> Which driver version?
Redmi K70 Ultra hyperOS2 Android 15 Input vulkaninfo Also Has This Phase Vk Extension （VK_EXT_robustness）

**Comment by SCP68** [comment]
> Which driver version?

![Screenshot_2024-12-09-14-01-47-726_com xiaomi vipaccount](https://github.com/user-attachments/assets/86cd6c72-8b80-4500-a74e-bc7c0fa93fe8)

So I'm going to choose to feed the relevant issues back to the Xiaomi community official hoping to get it to discuss with MediaTek and make it compatible

**Comment by qemuduewr** [comment]
> Which driver version?

Does mesa-vulkan-icd-wrapper support glibc?

**Comment by SCP68** [comment]
> > Which driver version?
> 
> Does mesa-vulkan-icd-wrapper support glibc?

mesa-vulkan-icd-wrapper (24.2.5-9)
